test/e2e_node/summary_test.go

```diff
@@ -90,7 +90,7 @@ var _ = framework.KubeDescribe("Summary API [NodeConformance]", func() {
 					"StartTime": recent(maxStartAge),
 					"CPU": ptrMatchAllFields(gstruct.Fields{
 						"Time":                 recent(maxStatsAge),
-						"UsageNanoCores":       bounded(10000, 2e9),
+						"UsageNanoCores":       bounded(0, 2e9),
```

Comment:
This change is related to flaky test behaviour - https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet/1313556070618107904

Reply:
Could you please add a comment of some sort explaining more in detail why this change is needed?
I see you added some sample output from /stats/summary in https://github.com/kubernetes/kubernetes/issues/95343#issuecomment-704748920 and it does show some pods with 0 but why would that be the case?

I've noticed this test is flaky in fedora and ubuntu. so my question is also: is there something wrong with the configuration of the test environment in fedore/ubuntu or is there some bad assumption that went into the current version of this test?
Ultimately, why should we check that the value of `UsageNanoCores` is between 0 and 2e9 instead of 10k and 2e9?

Reply:
It's unclear why 0 would be a valid state. Since it seems to be failing on the `gomega.Constantly` validation, it already passed the `gomega.Eventually` validation, meaning that the value was already greater than 10k. Is this covering up an issue? Or, is the idea that this is not the level of validation required for this API?

Looks like `UsageNanoCores` is zero when [cadvisor doesn't provide them.](https://github.com/kubernetes/kubernetes/blame/d19aec8bf1c8ca294345001d2286db578d422982/pkg/kubelet/stats/helper.go#L45)

Reply:
@alejandrox1 on fedora (and probably on ubuntu too) stats provider is CRI while on COS it might be cadvisor if they use docker. 

CRI stats provider tries to estimate the value of `UsageNanoCores` because CRI runtimes do not supply nano core usages. Check this [comment](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/cri_stats_provider.go#L106) on that function for more details. If any of these [conditions](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/cri_stats_provider.go#L667) are met then the value of the `UsageNanoCores` for that request won't be set. 

Similarly, If [this](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/cri_stats_provider.go#L533) call returns `nil`, we may never have the value for `UsageNanoCores` set. 

@mmerkes So the reason I set to 0, because for CRI stats provider the value can indeed be 0 unless the values in the cache are built up and [used in calculating](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/cri_stats_provider.go#L679) the effective `UsageNanoCores`

Reply:
I was just wondering, if it would help to play with the timeout values of `gomega.Constantly` and `gomega.Constantly` to see if that mitigates the issue.

Reply:
Just to add another point, I had 5 runs of this test on Fedora CoreOS by keeping the accepted values between 10000 and 2e9, and they all succeeded. I am not able to predicatibly reproduce this so far, pretty much like the behaviour we see on [test grid](https://k8s-testgrid.appspot.com/sig-release-master-blocking#node-kubelet-master).

Reply:
@harche could you document in someway what you said in https://github.com/kubernetes/kubernetes/pull/95345#discussion_r501630942 , please?
I think it makes sense, and this patch looks reasonable.
Plus it may be good to merge and see how the behaviour of the test changes.

Though at least for this test (just by looking at it) some of the conditions seem a tad arbitrary.
The reason behind this PR is interesting in some way and i think it would be good to leave a comment in the code detailing the why of this check.